STYLE_GUIDE.md

```diff
@@ -0,0 +1,383 @@
+# Swift Style Guide
```

Comment:
Did you add this whole thing, or did you just extract it from somewhere else? I wonder if there's any changes made to this.

Reply:
I extracted it and condensed it from `Best Practices`. There are definitely changes made to it to try and make it more concise and closer to enforcing what we're already doing.